src/caffe/test/test_net.cpp

```diff
@@ -2224,4 +2321,52 @@ TYPED_TEST(NetTest, TestReshape) {
   }
 }
 
+TYPED_TEST(NetTest, TestSkipPropagateDown) {
+  // check bottom_need_backward if propagate_down is true
+  this->InitSkipPropNet(false);
+  vector<bool> vec_layer_need_backward = this->net_->layer_need_backward();
+  for (int layer_id = 0; layer_id < this->net_->layers().size(); ++layer_id) {
+    if (this->net_->layer_names()[layer_id] == "loss") {
+      // access to bottom_need_backward coresponding to label's blob
+      bool need_back = this->net_->bottom_need_backward()[layer_id][1];
+      // if propagate_down is true, the loss layer will try to
+      // backpropagate on labels
+      CHECK_EQ(need_back, true)
+        << "bottom_need_backward should be True";
```

Comment:
All the `CHECK_EQ(.., true)`s in this test should be `EXPECT_TRUE` (and `CHECK_EQ(..., false)` -> `EXPECT_FALSE`) -- the tests should fail but not die if the behavior being tested is incorrect.